### DISCUSSION TITLE
waitUntil() timeout message should be shown on success too

### DISCUSSION THREAD

**Discussion by samuelfreiberg** [webdriverio/webdriverio#11634] [General]
I think there should be a way to display the timeoutMsg parameter on waitUntil() calls even when the condition passes (maybe green text for pass, red text for fail). This makes debugging so much easier. When looking through logs, it helps to see the flow of testing. If I'm taking the time to write a message, I would like to see it in the logs that it either passed or failed. If there could be an option for this, that would be so awesome.

**Comment by christian-bromann** [comment]
@samuelfreiberg could you explain your use case at an actual example? Remember you can always log things yourself through the `@wdio/logger` package.

**Reply by samuelfreiberg** [reply]
In my opinion, using @wdio/logger is redundant in this case. I look at waitUntil() as a sort of assert - "I **expect** this condition to be true in this timespan". **This is useful information when debugging! Even when it passes successfully.** It helps track the flow of UI testing when you see both successes and failures in the logs.

Let's take a simple example:

```
await browser.waitUntil(await async () => await Button.isExisting(),
{
  message: 'The button to open this test page exists!'
});
```

When I'm debugging, it's useful to see the state of the app at different points in testing. When I'm looking through the logs, I should be able to see this exact assert somewhere (pass or fail).


If I want to show that message now, I'd have to do this:
```
await browser.waitUntil(await async () => await Button.isExisting(),
{
  message: 'The button to open this test page exists!'
});
log.info( 'The button to open this test page exists!')l
```

**Reply by samuelfreiberg** [reply]
Also, let's say a test timed out (no asserts failed, just timeout). It's helpful to see all the successful waitUntil() actions to get a better idea of where in the testing in timed out (maybe a certain point of the test is taking too long - I don't know what part that is unless I dig into every little WDIO Fetch/Post call. Sure I could add the logging, but that's redundant with the waitUntil() messages).

**Reply by samuelfreiberg** [reply]
These are crucial points in the testing process, and the messages provide valuable information during debugging - pass or fail.

**Reply by samuelfreiberg** [reply]
Especially for engineers debugging failures that didn't write the tests (or have any clue about the tests). Think about an engineer trying to read through logs who don't have any knowledge of the test - seeing these success messages provide them insight on the state of testing at that point.

Again, this is possible with the @wdio/logger package, but it would literally be the same string I'm passing to the waitUntil() method.

**Reply by christian-bromann** [reply]
Where do you like this message being shown then?

**Reply by samuelfreiberg** [reply]
The exact same spot that it would show if it failed. But for a success, maybe the text is green, and test execution continues as normal.

**Reply by christian-bromann** [reply]
If a command fails, a test fails. In the stdout terminal we show test failures and not individual command states. You can use the `afterCommand` hook to log states at your will whenever a `waitUntil` is called. I am not sure if this is something that makes sense to enable by default.
